### ISSUE TITLE
Null pointer in the Editorial Reminder if the user has blocked notifications

### ISSUE THREAD

**Issue by nwoodward** [pkp/pkp-lib#10496]
**Describe the bug**
We see several errors in the OJS logs about a null pointer in EditorialReminder.php, and it's due to the $notification being null. In the handle() method below the $notification variable returned from `$notificationManager->createNotification()` could potentially be null. So when it's used in the `allowUnsubscribe()` method of `$mailable` it throws a null pointer error.

https://github.com/pkp/pkp-lib/blob/09388ce9f4f3feef99d52cdcf95517620bf50378/jobs/email/EditorialReminder.php#L143-L162

This happens when the user has blocked notifications. In the `createNotification()` method of `PKPNotificationOperationManager` there's a check at the beginning to see if the user exists and has blocked notifications, in which case `null` is returned.

https://github.com/pkp/pkp-lib/blob/09388ce9f4f3feef99d52cdcf95517620bf50378/classes/notification/PKPNotificationOperationManager.php#L143-L167

I don't understand the code well enough to say if the fix should be a null check on `$notification` after it gets created in `EditorialReminder.php`, and if so then return from `handle()`. Or if `$mailable` should still be called without the `allowUnsubscribe($notification)` method.

**What application are you using?**
 OJS 3.4.0-5

**PRs**

#### Stable-3.4.0
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/12275
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5306 [TEST ONLY]
[omp](https://github.com/pkp/omp) --> https://github.com/pkp/omp/pull/2224 [TEST ONLY]
[ops](https://github.com/pkp/ops) --> https://github.com/pkp/ops/pull/1193 [TEST ONLY]

#### Stable-3.5.0
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/12298
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5318 [TEST ONLY]
[omp](https://github.com/pkp/omp) --> https://github.com/pkp/omp/pull/2225 [TEST ONLY]
[ops](https://github.com/pkp/ops) --> https://github.com/pkp/ops/pull/1194 [TEST ONLY]

#### main
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/12299
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5319 [TEST ONLY]

**Comment by touhidurabir** [comment]
Fix is applied at https://github.com/pkp/pkp-lib/pull/10332/files#diff-83d61d197c61d91e16b32cc318edb9f2c3ba21032384d2337d75ce12c7189fdfR173-R177 for issue https://github.com/pkp/pkp-lib/issues/10214 .

**Comment by touhidurabir** [comment]
fixed by https://github.com/pkp/pkp-lib/issues/10214 .

**Comment by ajnyga** [comment]
This has not been ported to 3.4?

In 3.4.0.9 I still get these errors:
`    "message": "PKP\\mail\\mailables\\EditorialReminder::allowUnsubscribe(): Argument #1 ($notification) must be of type APP\\notification\\Notification, null given, called in lib\/pkp\/jobs\/email\/EditorialReminder.php on line 180",
`

**Comment by asmecher** [comment]
@touhidurabir, heads-up about the above question, thanks!

**Comment by touhidurabir** [comment]
Will take a look at this again, wondering from where the regression came .

**Comment by touhidurabir** [comment]
Re-opening this as there been a case missing for this 

we have only checked for case for blocked email notification (e.g. `blocked_emailed_notification`) . but if the notification itself is blocked (e.g. `blocked_notification`), the entry in the DB saved in the `notification_subscription_settings` as the `blocked_notification` , not as `blocked_emailed_notification` which cause getting a true (e.g. as subscribed ) for the following method in the job class 
```php
/**
 * Is this editor subscribed to this email notification type?
 */
protected function isSubscribed(): bool
{
    /** @var NotificationSubscriptionSettingsDAO $notificationSubscriptionSettingsDao  */
    $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
    $blockedEmails = $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
        NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
        $this->editorId,
        $this->contextId
    );

    return !in_array(Notification::NOTIFICATION_TYPE_EDITORIAL_REMINDER, $blockedEmails);
}
```

basically we need a check for the `NotificationSubscriptionSettingsDAO::BLOCKED_NOTIFICATION_KEY`, same as we have for `NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY` ;

**Comment by touhidurabir** [comment]
@kaitlinnewson @taslangraham can you please review the PRs at https://github.com/pkp/pkp-lib/issues/10496#issue-2562269626 . If all ok, I will forward port the PRs for 3.5 and main .

**Comment by kaitlinnewson** [comment]
Hi @touhidurabir, one question and a minor suggestion for a typo fix on the PR.

**Comment by touhidurabir** [comment]
@kaitlinnewson @taslangraham can you review the `pkp-lib` PRs for 3.5/main (basically forward porting from 3.4 branch) .

**Comment by touhidurabir** [comment]
Merged the `pkp-lib` PR for 3.4/3.5/main and confirming that all tests were passing for all 3 apps across all the branches .

**Comment by touhidurabir** [comment]
@Tribunal33 it's ready for testing but will be bit hard to tests as depends on the days duration and pending editorial task/action/decision .

**Comment by Tribunal33** [comment]
After speaking with @touhidurabir it doesn't look like this one is testable without artificially breaking it.  So we will defer this to be covered with a unit test in the future. 

I have created a placeholder ticket to improve the unit test coverage here : https://github.com/pkp/pkp-lib/issues/12345
